optimum/intel/openvino/modeling.py

```diff
@@ -1031,3 +1035,183 @@ def forward(self, **kwargs):
+        subfolder: str = "",
+        local_files_only: bool = False,
+        task: Optional[str] = None,
+        trust_remote_code: bool = False,
+        **kwargs,
+    ):
+        if use_auth_token is not None:
+            warnings.warn(
+                "The `use_auth_token` argument is deprecated and will be removed soon. Please use the `token` argument instead.",
+                FutureWarning,
+            )
+            if token is not None:
+                raise ValueError("You cannot use both `use_auth_token` and `token` arguments at the same time.")
+            token = use_auth_token
+            
+        # Fix the mismatch between timm_config and huggingface_config
+        local_open_clip_model = _is_open_clip_ov_dir(model_id)
+        if local_open_clip_model or (not os.path.isdir(model_id) and model_info(model_id).library_name == "open_clip"):
+            if not is_open_clip_available():
+                raise ImportError(
+                    "To load a timm model, timm needs to be installed. Please install it with `pip install timm`."
+                )
+
+            import open_clip
+
+            if local_open_clip_model:
+                config_path = os.path.join(model_id, "config.json")
+                config = {}
+                with open(config_path, 'r', encoding='utf-8') as f:
+                    config = json.load(f)
+
+                return super()._from_pretrained(model_id=model_id, config=config, file_name='openvino_text.xml')
+
+            model = OpenClipForZeroShotImageClassification.from_pretrained(model_id, cache_dir=cache_dir)
+            text_model = model.text
+            setattr(text_model, 'config', model.config.text_config)
+            text_cfg = OpenCLIPTextOpenVINOConfig(model.config.text_config, 'zero-shot-image-classification')
+            return cls._to_load(model=model.text, config=model.config.text_config, onnx_config=text_cfg, stateful=False, **kwargs)
+        else:
+            return OVModel.from_pretrained(
+                model_id=model_id,
+                config=config,
+                export=export,
+                token=token,
+                revision=revision,
+                force_download=force_download,
+                cache_dir=cache_dir,
+                subfolder=subfolder,
+                local_files_only=local_files_only,
+                task=task,
+                trust_remote_code=trust_remote_code,
+                **kwargs,
+            )
+
+    def forward(
+        self,
+        input_ids: Union[torch.Tensor, np.ndarray],
+        **kwargs,
+    ):
+        self.compile()
+
+        inputs = {
+            "text": input_ids
+        }
+
+        outputs = self._inference(inputs)
+        return ModelOutput(text_features=outputs["text_features"])
+
+
+@add_start_docstrings(
+    """
+    OpenVINO Model for OpenCLIP Vision model for tasks like zero-shot-image-classification.
+    """,
+    MODEL_START_DOCSTRING,
+)
+class OVModellCLIPVisual(OVModel):
```

Comment:
Should we also have class for ZeroShotClassification too that will use both text and vision models?

Reply:
I have added OVModellCLIPForZeroShotImageClassification with parent PreTrainedModel and text/vision_model which are OVBaseModel. Also I have implemented methods from OVBaseModel , but haven't added attributes such as output_names/input_names. Is it what expected ? Should I add consolidated model and choice what model user will use ?